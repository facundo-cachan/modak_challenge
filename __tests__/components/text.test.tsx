import { Text } from '@atoms'
import { render, screen } from '@testing-library/react-native'
import { withThemeProvider } from '../utils/customRender'

const text = 'Demo Text'

describe('Text component', () => {
  beforeEach(() => {
    render(withThemeProvider(<Text variant="large">{text}</Text>))
  })
  test('Render component', () => {
    expect(screen).toMatchSnapshot()
  })
  test('Accessibility', () => {
    const value = screen.getByRole('text')
    expect(value).toBeOnTheScreen()
    expect(value).toBeVisible()
    expect(value.props.accessibilityRole).toBe('text')
    expect(value.props.children).toBe(text)
    expect(value.props['aria-label']).toBe(text)
  })
  test('Visiblity & text value', () => {
    const value = screen.getByText(text)
    expect(value).toBeOnTheScreen()
    expect(value).toBeVisible()
    expect(value.props.children).toBe(text)
  })
})