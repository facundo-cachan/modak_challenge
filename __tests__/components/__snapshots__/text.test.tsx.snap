// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Render component 1`] = `
<Text
  accessibilityRole="text"
  aria-label="Demo Text"
  style={
    [
      {
        "flexWrap": "wrap",
      },
      {
        "color": "#115B99",
        "fontFamily": "Arial Hebrew",
        "fontSize": 33,
        "fontWeight": "bold",
        "lineHeight": 26.6,
        "textTransform": "capitalize",
      },
      undefined,
    ]
  }
>
  Demo Text
</Text>
`;
